eclass/coreos-kernel.eclass

```diff
@@ -123,9 +123,61 @@ kmake() {
 		"$@"
 }
 
-# Discard the module signing key, we use new keys for each build.
+# Prints the value of a given kernel config option.
+# Quotes around string values are removed.
+getconfig() {
+	local value=$(getfilevar_noexec "CONFIG_$1" build/.config)
+	[[ -n "${value}" ]] || die "$1 is in the kernel config"
+	[[ "${value}" == '"'*'"' ]] && value="${value:1:-1}"
+	echo "${value}"
+}
+
+# Generate the module signing key for this build.
+setup_keys() {
+	local sig_hash sig_key
+	sig_hash=$(getconfig MODULE_SIG_HASH)
+	sig_key="build/$(getconfig MODULE_SIG_KEY)"
+
+	if [[ "${sig_key}" == "build/certs/signing_key.pem" ]]; then
```

Comment:
Because we want to be using a key that the kernel build won't generate automatically. The private key is only available during the modules build so if the auto-generated key is used the kernel build would generate a new key pair in response to the missing private key. https://github.com/coreos/coreos-overlay/pull/2005/commits/9f37ede4bbc8843058ae5cbe6bcdf10539d3423e